### ISSUE TITLE
Use of yum in cloud boothook may cause problems

### ISSUE THREAD

**Issue by mes5k** [nextflow-io/nextflow#239]
This [line in AmazonCloudDriver](https://github.com/nextflow-io/nextflow/blob/master/src/main/groovy/nextflow/cloud/aws/AmazonCloudDriver.groovy#L161) uses yum to install some nfs utilities.  If the AMI used is based on a debian variant then this will likely cause problems.  I think a better approach would be simply require that the AMI have whatever actually nfs tools get used installed.

**Comment by pditommaso** [comment]
What do you think about adding the following?  

```
yum install -y nfs-utils || apt-get -y install nfs-common
```

**Comment by mes5k** [comment]
I guess that could work, but I'm still a bit uncomfortable having the package install happen behind the scenes in the boothook, while users are already expected to have done other package installs to create the AMI.  

A general issue with the boothook (not necessarily related to this ticket) is that it gets run in a way such that there are no logs generated.  I was able to see in /var/log/cloud-init.log that *something* had gone wrong in part-002, but not what the actual problem was. I had to run it manually to ultimately find the problem.

 I wonder if you could instead simply add a check for the existence of ```mount``` in cloud-boot.txt? That way if it's not there you should see an error in boot.log.

**Comment by pditommaso** [comment]
I agree that the user should be able to pre-install the nfs-utils, but at the same time NF should be able to install it if missing. 

The `mount` is available independently from the NFS client. It could be done something like this: 

```
which nfsstat >/dev/null  || yum install -y nfs-utils || apt-get -y install nfs-common 
```

**Comment by mes5k** [comment]
Yeah, I think that's a bit better.  If you want to do the nfs install here, that's your call!  :)

**Comment by pditommaso** [comment]
Included in version `0.22.5`
